weave/trace/serialize.py

```diff
@@ -57,6 +58,14 @@ def to_json(
             and not has_custom_repr(obj)
         ):
             return dictify(obj)
+
+        # TODO: I would prefer to only have this once in dictify? Maybe dictify and to_json need to be merged?
```

Comment:
I think it is mostly confusing we have the function `dictify` but conceptually `try_to_dict` is also a "dictify" of sorts. Maybe we could change `use_dictify` to something non-boolean, like "dictify_level" or "dictify_strategy".

Reply:
Conceptually I understand this as:
1. Functions may implement their own `to_dict`, which makes them `Dictifiable`
2. The current `dictify` is more of a "to_dict_of_last_resort" for objects that are not `Dictifiable`
3. To compose, rename the current `dictify->_to_dict_fallback`, then have a new func:
```py
def dictify(obj: Any) -> dict[str, Any]:
    if isinstance(obj, Dictifiable):
        return obj.to_dict()
    return _to_dict_fallback(obj)
```

Reply:
1. Yes
2. Yes, though it also handles the Dictifiable case.
3. The issue is the meaning of the `use_dictify` parameter. Tim is proposing we do `to_dict` even when use_dictify is false, a condition in which we wouldn't want to call the `_to_dict_fallback`